CMakePresets.json

```diff
@@ -0,0 +1,45 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "no-submodule",
+            "warnings": {
+                "dev": true
+            },
+            "cacheVariables": {
+                "OPENASSETIOTEST_ENABLE_SUBPROJECT": "OFF",
+                "OPENASSETIO_MEDIACREATIONTEST_ENABLE_SUBPROJECT": "OFF"
+            }
+        },
+        {
+            "name": "submodule-openassetio-find-mediacreation",
+            "warnings": {
+                "dev": true
+            },
+            "cacheVariables": {
+                "OPENASSETIOTEST_ENABLE_SUBPROJECT": "ON",
+                "OPENASSETIO_MEDIACREATIONTEST_ENABLE_SUBPROJECT": "OFF"
+            }
+        },
+        {
+            "name": "find-openassetio-submodule-mediacreation",
+            "warnings": {
+                "dev": true
+            },
+            "cacheVariables": {
+                "OPENASSETIOTEST_ENABLE_SUBPROJECT": "OFF",
+                "OPENASSETIO_MEDIACREATIONTEST_ENABLE_SUBPROJECT": "ON"
+            }
```

Comment:
[Question] Does the existence of the tests justify these odd presets existing as part of the project. One could argue they serve as a pretty poor example.

Reply:
Yeah I'd be tempted to set these flags on top of any base preset (which it looks like we don't have really, so could just remove), I think it would read clearer in CI.

Reply:
Aye, got rid of the presets entirely. 

https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/7/commits/1615b4a79a5dffb8662bd3b040ed71c80e5fac49
https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/7/commits/2e729d995724aae942a9bd98c5dcda1c27557e07